### PR TITLE
PIPRES-6: added search and filter for subscriptions in admin page

### PR DESCRIPTION
**Pull request by margud**
Added subscription search and filter. Fixed filter errors. Also tried to fix automated test but with no luck. for now skipping it and will come back when main feature is done.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -32,10 +27,11 @@ jobs:
       - run: composer i
 
       - name: Pull PrestaShop files (Tag 1.7.7.0)
-        run: docker run -tid --rm -v ps-volume:/var/www/html -v $PWD/:/var/www/html/modules/mollie --name temp-ps prestashop/prestashop:1.7.7.0
+        run: docker run -tid --rm -v ps-volume:/var/www/html -v $PWD/:/var/www/html/modules/mollie --name temp-ps prestashop/prestashop:1.7.8.1-7.4-apache
```
Comment (margud) [inline .github/workflows/test.yml:30]: ps, tests on github not working right now. tried to fix it but took to much time so added task to fix it after main feature is done.

File: subscription/Grid/SubscriptionGridQueryBuilder.php [hunk 2]
```diff
@@ -100,28 +98,66 @@ private function getQueryBuilder(array $filters): QueryBuilder
...
 
         foreach ($filters as $filterName => $value) {
-            if (!array_key_exists($filterName, $allowedFiltersMap)) {
+            if ('fullname' === $filterName) {
+                $qb->andWhere($likeComparisonFilters[$filterName] . ' LIKE :' . $filterName)
+                    ->setParameter($filterName, '%' . $value . '%');
                 continue;
             }
 
             if ('id_attribute_subgroup' === $filterName) {
```
Comment (mandan2) [inline subscription/Grid/SubscriptionGridQueryBuilder.php:125]: This key was deleted in array above, so statement won't be true.

File: subscription/Grid/SubscriptionGridQueryBuilder.php [hunk 3]
```diff
@@ -100,28 +98,66 @@ private function getQueryBuilder(array $filters): QueryBuilder
...
                 continue;
             }
 
-            $qb->andWhere($allowedFiltersMap[$filterName] . ' LIKE :' . $filterName)
-                ->setParameter($filterName, '%' . $value . '%')
-            ;
+            if (array_key_exists($filterName, $dateComparisonFilters)) {
+                $alias = $dateComparisonFilters[$filterName];
+
+                foreach ($value as $name => $dateValue) {
```
Comment (mandan2) [inline subscription/Grid/SubscriptionGridQueryBuilder.php:140]: Have you tested edge cases? E.g. select from date to be same as some record date, same for end date.

File: subscription/Grid/SubscriptionGridQueryBuilder.php [hunk 4]
```diff
@@ -100,28 +98,66 @@ private function getQueryBuilder(array $filters): QueryBuilder
...
+            }
         }
     }
+
+    /**
+     * @return string
+     */
+    private function getNameField(): string
+    {
+        return 'CONCAT(customer.firstname, " ", customer.lastname)';
```
Comment (mandan2) [inline subscription/Grid/SubscriptionGridQueryBuilder.php:161]: Utility could be used to join two strings

**Review by mandan2 — CHANGES_REQUESTED** [review]

File: subscription/Grid/SubscriptionGridQueryBuilder.php (hunk 3 above)
Comment (margud) [reply subscription/Grid/SubscriptionGridQueryBuilder.php:140]: yes tested it

File: subscription/Grid/SubscriptionGridQueryBuilder.php (hunk 4 above)
Comment (margud) [reply subscription/Grid/SubscriptionGridQueryBuilder.php:161]: look again :D this is one string. we add this into mysql to call concat that joins 2 strings using mysql

**Review by mandan2 — APPROVED** [review]

**Pull request merged**